examples/plot_isotonic_regression.py

```diff
@@ -3,13 +3,20 @@
 Isotonic Regression
 ===================
 
-An illustration of the isotonic regression on generated data. The
-isotonic regression finds a non-decreasing approximation of a function
-while minimizing the mean squared error on the training data. The benefit
-of such a model is that it does not assume any form for the target
+An illustration of the isotonic regression on generated data (non-linear
+monotonic trend with homoschedastic uniform noise).
+
+The isotonic regression algorithm finds a non-decreasing approximation of a
+function while minimizing the mean squared error on the training data. The
+benefit of such a model is that it does not assume any form for the target
 function such as linearity. For comparison a linear regression is also
 presented.
 
+The plot on the right-hand side shows the model prediction function that
+results from the linear interpolation of thresholds points. The thresholds
+points are a subset of the training input observations and their matching
+target values as computed by the isotonic non-parametric fit.
```

Comment:
The current formulation suggests that the predicted values are values from the training set which I believe isn't true (I might be wrong)

```suggestion
target values are computed by the isotonic non-parametric fit.
```